### PR TITLE
[Android] Log HDCP Level

### PR DESCRIPTION
**Pull request by borongc**
As 4k drm requires hdcp 2.2, we log the hdcp level for debugging.

b/307430294

### CODE REVIEW

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java [hunk 1]
```diff
@@ -373,6 +373,37 @@ private MediaDrmBridge(String keySystem, UUID schemeUUID, long nativeMediaDrmBri
...
+    switch (mMediaDrm.getConnectedHdcpLevel()) {
+      case MediaDrm.HDCP_V1:
+      case MediaDrm.HDCP_V2:
+      case MediaDrm.HDCP_V2_1:
+        Log.i(
+            TAG,
+            String.format(
+                "MediaDrm HDCP Level is < HDCP_V2_2 (%d).",
+                mMediaDrm.getConnectedHdcpLevel()));
+        break;
```
Comment (xiaomings) [inline starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java:386]: We should log them individually, without mentioning whether they are less than HDCP_V2_2, just as how we log HDCP_NO_DIGITAL_OUTPUT below.

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java [hunk 2]
```diff
@@ -373,6 +373,37 @@ private MediaDrmBridge(String keySystem, UUID schemeUUID, long nativeMediaDrmBri
...
+      case MediaDrm.HDCP_NONE:
+        Log.i(TAG, "MediaDrm HDCP Level is HDCP_NONE.");
+        break;
+      case MediaDrm.HDCP_NO_DIGITAL_OUTPUT:
+        Log.i(TAG, "MediaDrm HDCP Level is HDCP_NO_DIGITAL_OUTPUT.");
+        break;
+      case MediaDrm.HDCP_LEVEL_UNKNOWN:
+      default:
+        Log.i(TAG, "MediaDrm HDCP Level is HDCP_LEVEL_UNKNOWN.");
```
Comment (xiaomings) [inline starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java:403]: Log them separately, and in the default case, also log the value.

Comment (borongc) [reply starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java:403]: Fixed.

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java (hunk 1 above)
Comment (borongc) [reply starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java:386]: Fixed.

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java [hunk 3]
```diff
@@ -373,6 +373,37 @@ private MediaDrmBridge(String keySystem, UUID schemeUUID, long nativeMediaDrmBri
...
+      case MediaDrm.HDCP_V2:
+        Log.i(TAG, "MediaDrm HDCP Level is HDCP_V2.");
+        break;
+      case MediaDrm.HDCP_V2_1:
+        Log.i(TAG, "MediaDrm HDCP Level is HDCP_V2_1.");
+        break;
+      case MediaDrm.HDCP_V2_2:
+        Log.i(TAG, "MediaDrm HDCP Level is HDCP_V2_2.");
+        break;
+      case MediaDrm.HDCP_V2_3:
```
Comment (xiaomings) [inline starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java:390]: It seems that most of the constants are available in api level 28 or 29 (e.g. https://developer.android.com/reference/android/media/MediaDrm#HDCP_V2_2).  We can either guard them individually using the api level they are first available, or guard the whole block using the first api level they are all available.

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java [hunk 4]
```diff
@@ -373,6 +373,37 @@ private MediaDrmBridge(String keySystem, UUID schemeUUID, long nativeMediaDrmBri
     mSchemeUUID = schemeUUID;
     mMediaDrm = new MediaDrm(schemeUUID);
 
+    // Get info of hdcp connection, as 4k drm require hdcp 2.2
```
Comment (xiaomings) [inline starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java:376]: Consider removing the mentioning of 4k and hdcp 2.2, as the code below isn't specific for this purpose.

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java [hunk 5]
```diff
@@ -373,6 +373,37 @@ private MediaDrmBridge(String keySystem, UUID schemeUUID, long nativeMediaDrmBri
...
+        Log.i(TAG, "MediaDrm HDCP Level is HDCP_NONE.");
+        break;
+      case MediaDrm.HDCP_NO_DIGITAL_OUTPUT:
+        Log.i(TAG, "MediaDrm HDCP Level is HDCP_NO_DIGITAL_OUTPUT.");
+        break;
+      case MediaDrm.HDCP_LEVEL_UNKNOWN:
+        Log.i(TAG, "MediaDrm HDCP Level is HDCP_LEVEL_UNKNOWN.");
+        break;
+      default:
+        Log.i(TAG, String.format("MediaDrm HDCP Level is %d.", mMediaDrm.getConnectedHdcpLevel()));
```
Comment (xiaomings) [inline starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java:403]: nit:
1. Use a variable at line 377, so we don't have to call the same function twice.
2. Rephrase the message to something like "Unknown MediaDrm HDCP level %d ...", i.e. explicitly mentioning unknown here.

Comment (borongc) [reply starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java:403]: Fixed.

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java (hunk 3 above)
Comment (borongc) [reply starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java:390]: Fixed.

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java (hunk 4 above)
Comment (borongc) [reply starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java:376]: Fixed.

**Review by xiaomings — APPROVED** [review]

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java [hunk 6]
```diff
@@ -373,6 +373,11 @@ private MediaDrmBridge(String keySystem, UUID schemeUUID, long nativeMediaDrmBri
     mSchemeUUID = schemeUUID;
     mMediaDrm = new MediaDrm(schemeUUID);
 
+    // Get info of hdcp connection
+    if (Build.VERSION.SDK_INT >= 28) {
```
Comment (xiaomings) [inline starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java:377]: This has to be 29, as HDCP_V2_3 is only available in 29.

Comment (borongc) [reply starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaDrmBridge.java]: Fixed.

**Review by xiaomings — APPROVED** [review]

**Pull request merged**